### PR TITLE
added onhttprequest event hook to enable blazor support

### PR DESCRIPTION
**Pull request by acquleo**
A OnHttpRequest event is raised before the request in order to be able to execute the SetBrowserResponseStreamingEnabled method on a Blazor app client.

### CODE REVIEW

**Review by eli-darkly — CHANGES_REQUESTED** [review]
Providing a hook for general modification of request attributes is a reasonable feature. However, I think the `event` pattern is not a great fit for this. All of the other events defined on `EventSource` are for _notifications_ of things that have happened— not for making changes to the behavior of `EventSource`. And an `event` inherently can have any number of handlers, which isn't a good fit for this feature since the ordering of event handler calls is not well defined but the ordering of changes to an HTTP request could be significant.

So, instead of exposing an `event` for this, I would rather provide a `ConfigurationBuilder` setter: `RequestModifier(Action<HttpRequestMessage>)`. If this hook is non-null, it is called by `EventSourceService` at the same place that you are currently calling `OnHttpRequest?.Invoke`.

**Comment by acquleo** [comment]
I've modified the code according to your proposal.

File: src/LaunchDarkly.EventSource/Configuration.cs [hunk 1]
```diff
@@ -190,6 +196,9 @@ internal Configuration(ConfigurationBuilder builder)
             RequestHeaders = new Dictionary<string, string>(builder._requestHeaders);
             ResponseStartTimeout = builder._responseStartTimeout;
             RequestBodyFactory = builder._requestBodyFactory;
+            HttpRequestModifier = builder._httpRequestModifier; 
+
+
```
Comment (eli-darkly) [inline src/LaunchDarkly.EventSource/Configuration.cs:201]: Minor: extra blank lines here.

File: src/LaunchDarkly.EventSource/Configuration.cs [hunk 2]
```diff
@@ -93,6 +93,12 @@ public sealed class Configuration
         /// <seealso cref="ConfigurationBuilder.HttpMessageHandler(HttpMessageHandler)"/>
         public HttpMessageHandler HttpMessageHandler { get; }
 
+        /// <summary>
+        /// Delegate hook invoked before an http request has been performed.
```
Comment (eli-darkly) [inline src/LaunchDarkly.EventSource/Configuration.cs:97]:
```suggestion
        /// Delegate hook invoked before an HTTP request has been performed.
```

File: src/LaunchDarkly.EventSource/ConfigurationBuilder.cs [hunk 3]
```diff
@@ -77,6 +77,18 @@ internal ConfigurationBuilder(Uri uri)
         public ConfigurationBuilder ConnectionTimeout(TimeSpan responseStartTimeout) =>
             ResponseStartTimeout(responseStartTimeout);
 
+
+        /// <summary>
+        /// Sets a delegate hook invoked before an http request has been performed.
```
Comment (eli-darkly) [inline src/LaunchDarkly.EventSource/ConfigurationBuilder.cs:82]:
```suggestion
        /// Sets a delegate hook invoked before an HTTP request has been performed, allowing changes to the request.
```

File: src/LaunchDarkly.EventSource/EventSource.cs [hunk 4]
```diff
@@ -302,6 +302,7 @@ private async Task ConnectToEventSourceAsync(CancellationToken cancellationToken
 
             var svc = GetEventSourceService(_configuration);
 
+
```
Comment (eli-darkly) [inline src/LaunchDarkly.EventSource/EventSource.cs:305]: Extra blank line.

File: src/LaunchDarkly.EventSource/LaunchDarkly.EventSource.csproj [hunk 5]
```diff
@@ -49,5 +49,6 @@
 
   <PropertyGroup>
     <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\LaunchDarkly.EventSource.xml</DocumentationFile>
+    <AssemblyOriginatorKeyFile>../../LaunchDarkly.EventSource.snk</AssemblyOriginatorKeyFile>
```
Comment (eli-darkly) [inline src/LaunchDarkly.EventSource/LaunchDarkly.EventSource.csproj:52]: I don't know why this change was made, but please revert it. We do not want the assembly to be strong-named in _every_ configuration, but only in the Release configuration, as is specified on line 33 of this file. Possibly your IDE made this change automatically for some reason, but please be careful that you're not checking in unintended changes to the project configuration.

**Review by eli-darkly — CHANGES_REQUESTED** [review]

**Review by acquleo — COMMENTED** [review]
agree with your comments

**Comment by eli-darkly** [comment]
This feature is now in the 4.2.0 release. Sorry it took so long for us to get back to it.

**Pull request merged**